enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,244 @@
+enhancement proposal document to learn more about WMCO upgrade.
+
+The procedure for an upgrade is as follows:
+1) MachineSet
+   * As part of upgrade Machine object gets deleted, resulting in the drain and deletion of the Windows node.
+   * New Windows VM will be created.
+   * The upgraded WMCO instance will configure Windows VM with containerd as the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+2) BYOH ( Bring Your Own Host)
+   * upgrade process will uninstall kubelet, kube-proxy, CNI, and hybrid-overlay components that were installed by WMCO.
+   * Windows OS-specific configurations like HNS-Network that were created by WMCO instance would be deleted.
+   * Once cleanup is done, WMCO instance will install containerd along with kubelet, kube-proxy, CNI and
+     hybrid-overlay.
+   * Kubelet will start as a service. Containerd will be the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed because Docker runtime support won't be available in kubelet. We work closely with the
+  containerd open source community and try to resolve it as soon as possible. We aim to fix any critical issue in the
+  upstream first and then update the containerd submodule to point to the latest containerd upstream version.
+  As we don't support downgrade, reverting to an older version is not possible. To overcome
+  any issues, we introduce this feature as part of the 5.0 community operator so that this feature
+  can be well tested before we make it default in WMCO 6.0 release. We will use two test jobs to test all e2e test cases
+  in the 4.10 community branch(OKD) and 4.10 branch(ccm OCP)to capture any regression.
+  This will help us to identify any issues well before containerd becomes default runtime.
+* containerd doesn't support image-pull-progress-deadline as of now. There is a PR
+  https://github.com/containerd/containerd/pull/6150 work in progress. Until this
+  gets merged, if Windows image pull takes more time than the default value, we might
+  run into to image pull timeout error. To overcome this issue, the user should log in to Windows VM and install `ctr or crictl`
+  command-line tool. Users can use one of these tools to pull the container image manually. Upon successful image download,
+  the user can create pods. Customers also can bake required container images into Windows VM image.
+* Currently Windows_exporter has been used to collect metrics from Windows node. We do
+  have containerd support in https://github.com/prometheus-community/Windows_exporter/releases/tag/v0.16.0
+  As of now, there is no feature parity mismatch between Docker and containerd.
+
```

Comment:
I see you have removed `If we run into any issue in containerd, we work closely with the containerd open source community and try to resolve it as soon as possible.`. This is important with a mention about carry patches.

I see your [point](https://github.com/openshift/enhancements/pull/962#discussion_r764371219) but there have been situations with k/k that has forced o/k to include carry patches. While this must be avoided as much as possible but sometimes it is the reality.

Reply:
I combined them in the first statement. 
>> and the new version introduces an issue due to containerd
  that should be addressed because Docker runtime support won't be available in kubelet. We work closely with the
  containerd open source community and try to resolve it as soon as possible.